server/src/routes/api/v2/contacts.js

```diff
@@ -0,0 +1,69 @@
+const router = require('express').Router()
+const mongoose = require('mongoose')
+const { ObjectId } = mongoose.Types
+const auth = require('@routes/auth')
+const UserWallet = mongoose.model('UserWallet')
+const Contact = mongoose.model('Contact')
+const { concat, compact, indexOf } = require('lodash')
+
+/**
+ * @api {post} /contacts/ Sync contacts list
+ * @apiName SyncContacts
+ * @apiGroup Contacts
+ * @apiDescription Sync contacts list
+ *
+ * @apiHeader {String} Authorization JWT Authorization in a format "Bearer {jwtToken}"
+ *
+ * @apiParam {String[]} contacts phone numbers list
+ *
+ * @apiSuccess {Object[]} new contacts list (phoneNumber, account)
+ * @apiSuccess {Number} nonce
+ */
+router.post('/', auth.required, async (req, res) => {
+  const { phoneNumber, accountAddress } = req.user
```

Comment:
Why do we needs the `accountAddress` for?
Do a user defined by its `phoneNumber` only or with its `accountAddress` also?
If this is to address the issue of a restoring the phone user, then we do need to solve it but with `resetSync` method called in the onboarding that simply remove the contacts set.

Reply:
It's not related to this feature. This is just how a [UserWallet](https://github.com/fuseio/fuse-studio/blob/master/server/src/models/UserWallet.js) is defined in the system.